### PR TITLE
Feat: update data according to params in url

### PR DESCRIPTION
**Pull request by MertsAnastasiya**

### CODE REVIEW

File: src/modules/app/app.ts [hunk 1]
```diff
@@ -45,20 +47,38 @@ export class App {
...
+                        this.globalFiltres.setCurrentSliders(key as SliderType, objectMinMax);
+                    } else if (key === 'category' || key === 'brand') {
+                        searchParams[key]!.split('|').forEach((item) => {
+                            const checkbox =  document.getElementById(item!)! as HTMLInputElement;
+                            checkbox.checked = true;
+                            this.globalFiltres.setCurrentFilters(key as FilterType, item);
+                        })
+                    }
+                }
+            }
```
Comment (alexey-ulashchick) [inline src/modules/app/app.ts:73]: It's ok for now if it works. In real life I would suggest move this to function and cover with tests. So much going on here..

Comment (alexey-ulashchick) [inline src/modules/app/app.ts:73]: It's ok for now if it works. In real life I would suggest move this to function and cover with tests. So much going on here..

File: src/modules/app/app.ts [hunk 2]
```diff
@@ -45,20 +47,38 @@ export class App {
...
+            this.updateResult(this.globalFiltres.getCurrentFilters(), this.globalFiltres.getCurrentSliders(), searchParams['search'] || '');
+            const button = new Button(
+                document.querySelector('.buttons__wrapper')!,
+                (type: string) => this.onClickButton(type)
+            );
+            button.drawButton('copy');
+            button.drawButton('reset');
```
Comment (alexey-ulashchick) [inline src/modules/app/app.ts:80]: Looks weird. Do we need to call twice with 2 different parameters?

Comment (alexey-ulashchick) [inline src/modules/app/app.ts:80]: Looks weird. Do we need to call twice with 2 different parameters?

File: src/modules/productPage.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        description!.innerHTML = this.product.description;
+        addCartButton.setAttribute('id', this.product.id.toString());
+        price!.innerHTML = `€${this.product.price}`;
+        rating!.innerHTML = `Rating: ${this.product.rating}`;
+        if (this.product.images[0]) {
+            mainImg.src = this.product.images[0]!;
+        }
+
+        let i = 0;
+        while (i !== 3) {
```
Comment (alexey-ulashchick) [inline src/modules/productPage.ts:68]: As a unfamiliar reader: what is 3? Why everything except 3? 
Move to constant at least and probably change to `<= 3`

**Review by alexey-ulashchick — COMMENTED** [review]
LGTM

File: src/modules/app/app.ts (hunk 2 above)
Comment (MertsAnastasiya) [reply src/modules/app/app.ts:80]: We need to create two buttons with different functionality.
I don't know how I can create two buttons in other way.

File: src/modules/productPage.ts (hunk 3 above)
Comment (MertsAnastasiya) [reply src/modules/productPage.ts]: I used 3 for specifying max image count.
Moved it to const.

File: src/modules/app/app.ts (hunk 2 above)
Comment (alexey-ulashchick) [reply src/modules/app/app.ts:80]: When you need 2 button, I would expect to see 2 objects. 
```ts
const button1: Button = new Button('');
const button2: Button = new Button('');
```
Here I see same button drawn twice? Redrawn? Something wrong here...

Comment (MertsAnastasiya) [reply src/modules/app/app.ts:80]: Get it, thanks

Comment (MertsAnastasiya) [reply src/modules/app/app.ts]: Done

File: src/modules/app/app.ts [hunk 4]
```diff
@@ -29,36 +34,97 @@ export class App {
...
+            return;
+        }
+
         this.globalFiltres.createFilters(productsData);
+        const searchParams = Object.fromEntries(new URLSearchParams(search));
 
-        const startFilters: Map<FilterType, Set<string>> = this.globalFiltres.getCurrentFilters();
-        const startSliders: Map<SliderType, SliderValue> = this.globalFiltres.getCurrentSliders();
-        this.updateResult(startFilters, startSliders, '');
+        if (search) {
```
Comment (alexey-ulashchick) [inline src/modules/app/app.ts:71]: When parameter is not `boolean` I would suggest this type of `if`. Is it comparison against `null`, `undefined`, `empty` or something else? Make it explicit.

File: src/modules/filters/globalFilters.ts [hunk 5]
```diff
@@ -155,4 +174,24 @@ export class GlobalFilters {
...
     }
+
+    private getMax(data: IProduct[], property: keyof IProduct): number {
+        let max: number = 0;
+        data.forEach(element => {
+            if(element[property] > max) {
+                max = Number(element[property]);
+            }
+        });
+        return max;
```
Comment (alexey-ulashchick) [inline src/modules/filters/globalFilters.ts:185]:
```suggestion
        return Math.max(...data.map(element => Number(element[property]));
```

**Review by alexey-ulashchick — APPROVED** [review]
LGTM

File: src/modules/app/app.ts (hunk 1 above)
Comment (MertsAnastasiya) [reply src/modules/app/app.ts]: Fixed.

File: src/modules/app/app.ts (hunk 4 above)
Comment (MertsAnastasiya) [reply src/modules/app/app.ts:71]: Fixed.

File: src/modules/filters/globalFilters.ts (hunk 5 above)
Comment (MertsAnastasiya) [reply src/modules/filters/globalFilters.ts:185]: Fixed.

**Pull request merged**